ProteoformSuiteInternal/Lollipop.cs

```diff
@@ -722,6 +722,24 @@ public List<ExperimentalProteoform> createProteoforms(IEnumerable<NeuCodePair> r
             return candidateExperimentalProteoforms;
         }
 
+        public bool consecutive_charge_states(int num_charge_states, List<ChargeState> charge_states)
+        {
+            List<int> charges = charge_states.OrderBy(cs => cs.charge_count).Select(cs => cs.charge_count).ToList();
+            if (charges.Count < num_charge_states) return false;
+            foreach(int cs in charges)
+            {
+                int consecutive = 1;
+                foreach(int next in charges)
+                {
+                    if (consecutive >= num_charge_states) return true;
+                    if (next == cs || next < cs) continue;
+                    if (next - cs == charges.IndexOf(next) - charges.IndexOf(cs)) consecutive++;
```

Comment:
Why are you comparing difference in charges to difference in indices of a list?

Reply:
I think was so I could keep checking - if I had charge states 4, 5, 6, then 6 -4 is 2 and the indexes are 2 apart so they're consecutive